apps/examples/fs_helper/fs_rw_test.c

```diff
@@ -0,0 +1,221 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/time.h>
+#include <fcntl.h>
+#include <errno.h>
+
+/****************************************************************************
+ * Definitions
+ ****************************************************************************/
+#define USAGE															\
+	"\n Filesystem R/W Test\n"											\
+	"\n usage: fs_rw_test [Path] [Size] [Iteration] [Operation] [Log]\n" \
+	"\n [Path] : Test File Path\n"										\
+	" [Size] : Data size\n"												\
+	" [Iteration] : Number of repetitions\n"							\
+	" [Operation] : 1.Read/Write / 2.Read / 3.Write\n"					\
+	" [Log] : Log Print Option [y/n]\n"									\
+	"\n ex) fs_rw_test /mnt/test 128 30 1 y\n"
+
+/****************************************************************************
+ * global value
+ ****************************************************************************/
+
+int total_time;
+
+/****************************************************************************
+ * fs_rw_test
+ ****************************************************************************/
+
+static inline long get_time(void)
+{
+	struct timeval tv;
+
+	gettimeofday(&tv, NULL);
+	return (tv.tv_sec * 1000 + tv.tv_usec / 1000);
+}
+
+static void rw_testlog(char *printopt, const char *format, ...)
+{
+	va_list ap;
+
+	if (printopt[0] == 'y' || printopt[0] == 'Y') {
+		va_start(ap, format);
+		vfprintf(stdout, format, ap);
+		va_end(ap);
+	}
+}
+
+static int t_open(const char *path, int oflags, char *printopt)
+{
+	int fd = -1;
+	long t = 0;
+	long dt = 0;
+	char mode = 0;
+
+	if ((oflags & O_RDONLY) != 0) {
+		mode = 'r';
+	} else if ((oflags & O_WRONLY) != 0) {
+		mode = 'w';
+	} else {
+		printf("open fail please select O_RDONLY or O_WRONLY option\n");
+		return -1;
+	}
+
+	t = get_time();
+	fd = open(path, oflags);
+	if (fd == -1) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		exit(EXIT_FAILURE);
+	}
+	dt = get_time() - t;
+	total_time += dt;
+	rw_testlog(printopt, "open(%c):%dms, ", mode, dt);
+	return fd;
+}
+
+static void t_write(int fd, const void *buf, size_t nbytes, char *printopt)
+{
+	long t = 0;
+	long dt = 0;
+	int ret = 0;
+
+	t = get_time();
+	ret = write(fd, buf, nbytes);
```

Comment:
You'd better to check not only it is -1 but also it is same to nbyte that you want to write.
On API document, it returns written size on file and sometimes it doesn't match each other.

Reply:
I modifed